cmd/install.go

```diff
@@ -278,378 +272,427 @@ func runInstallDupeDetectionSubCommand(ctx context.Context, config *configs.Conf
+		setupComponentWorkingEnvironment(ctx, config, "rqservice", "rqservice.toml", toolConfig)
+
+	default:
+		log.WithContext(ctx).Warn("Please select correct tool type!")
+		return nil
+	}
+
 	return nil
 }
 
-func uncompressNodeArchive(ctx context.Context, dstFolder string, archiveFile string) error {
-	file, err := os.Open(archiveFile)
+func processArchive(ctx context.Context, dstFolder string, archivePath string) error {
+	log.WithContext(ctx).Debugf("Extracting archive files from %s to %s", archivePath, dstFolder)
+
+	file, err := os.Open(archivePath)
 	if err != nil {
-		log.WithContext(ctx).Error("Not found archive file!!!")
+		log.WithContext(ctx).WithError(err).Errorf("Not found archive file - %s", archivePath)
 		return err
 	}
 	defer file.Close()
-
-	_, err = utils.Unzip(archiveFile, dstFolder)
-
+	_, err = utils.Unzip(archivePath, dstFolder)
 	if err != nil {
-		log.WithContext(ctx).Error("Extracting pastel executables Error!!!")
+		log.WithContext(ctx).WithError(err).Errorf("Failed to extract executables from %s", archivePath)
+		return err
+	}
+	log.WithContext(ctx).Debug("Delete archive files")
+	if err = utils.DeleteFile(archivePath); err != nil {
+		log.WithContext(ctx).Errorf("Failed to delete archive file : %s", archivePath)
 		return err
 	}
 
 	return nil
 }
 
-func uncompressArchive(ctx context.Context, dstFolder string, archiveFile string) error {
-	file, err := os.Open(archiveFile)
-
-	if err != nil {
-		log.WithContext(ctx).Error("Not found archive file!!!")
-		return err
+func makeExecutable(ctx context.Context, dirPath string, fileName string) error {
+	if utils.GetOS() == constants.Linux {
+		if _, err := RunCMD("chmod", "777", filepath.Join(dirPath, fileName)); err != nil {
+			log.WithContext(ctx).Errorf("Failed to make %s as executable", fileName)
+			return err
+		}
 	}
-	defer file.Close()
+	return nil
+}
 
-	_, err = utils.Unzip(archiveFile, dstFolder)
+func setupComponentWorkingEnvironment(ctx context.Context, config *configs.Config,
+	toolName string, configFileName string, toolConfig string) error {
 
-	if err != nil {
+	log.WithContext(ctx).Infof("Initialize working directory for %s", toolName)
+
+	workDirPath := filepath.Join(config.WorkingDir, toolName)
+	if err := utils.CreateFolder(ctx, workDirPath, config.Force); err != nil {
+		log.WithContext(ctx).Errorf("Failed to %s directory ", workDirPath)
 		return err
 	}
-	return nil
-}
 
-// InitializeFunc - Initialize the function
-func InitializeFunc(ctx context.Context, config *configs.Config) (err error) {
-	configJSON, err := config.String()
+	fileName, err := utils.CreateFile(ctx, filepath.Join(workDirPath, configFileName), config.Force)
 	if err != nil {
+		log.WithContext(ctx).Errorf("Failed to create %s file", configFileName)
 		return err
 	}
 
-	if err = config.SaveConfig(); err != nil {
+	if err = utils.WriteFile(fileName, toolConfig); err != nil {
+		log.WithContext(ctx).Errorf("Failed to write config to %s file", configFileName)
```

Comment:
nit: should include error in same log, than log error in caller function - easy to associate error with command than.
or don't need to log here. simpy wrap error with context, like:
return errors.Errorf("failed to write config to % file : %v", configFileName, err)